src/overlays/actors/ovl_Boss_02/z_boss_02.c

```diff
@@ -1630,166 +1660,194 @@ void Boss02_DrawEffects(PlayState* play) {
+                goto done;
             }
-            sp57 = 1;
-            goto block_38;
+            break;
 
-        case 10:
-            if ((this->unk_1D14 < 30) && (D_809E0421 != 0) &&
+        case GIANTS_MASK_CS_STATE_MASK_OFF:
+            if ((this->giantsMaskCsTimer < 30) && sCanSkipMaskOffCs &&
                 CHECK_BTN_ANY(CONTROLLER1(&play->state)->press.button,
                               BTN_A | BTN_B | BTN_CUP | BTN_CDOWN | BTN_CLEFT | BTN_CRIGHT)) {
-                this->unk_1D18++;
-                this->unk_1D78 = 1;
-                this->unk_1D14 = 0;
+                this->giantsMaskCsState++;
+                this->giantsMaskCsFlashState = GIANTS_MASK_CS_FLASH_STATE_STARTED;
+                this->giantsMaskCsTimer = 0;
                 break;
             }
 
-        label2:
-            if (this->unk_1D14 != 0) {
-                if (this->unk_1D14 == (u32)(BREG(44) + 10)) {
+        maskOff:
+            if (this->giantsMaskCsTimer != 0) {
+                if (this->giantsMaskCsTimer == (u32)(BREG(44) + 10)) {
                     Audio_PlaySfx(NA_SE_PL_TRANSFORM_NORAML);
                 }
-                Math_ApproachF(&this->unk_1D64, 60.0f, 0.1f, this->subCamAtVel * 640.0f);
-                Math_ApproachF(&this->unk_1D6C, 23.0f, 0.1f, this->subCamAtVel * 150.0f);
-                Math_ApproachF(&this->unk_1D70, 0.01f, 0.1f, 0.003f);
-                Math_ApproachF(&this->unk_1D74, 0.0f, 1.0f, this->subCamAtVel * 100.0f);
+
+                // Makes the camera move slowly towards the player and look downwards as the player
+                // decreases in size. Also increases the fogNear from -100.0f to 0.0f.
+                Math_ApproachF(&this->subCamDistZFromPlayer, 60.0f, 0.1f, this->subCamAtVel * 640.0f);
+                Math_ApproachF(&this->subCamAtOffsetY, 23.0f, 0.1f, this->subCamAtVel * 150.0f);
+                Math_ApproachF(&this->playerScale, 0.01f, 0.1f, 0.003f);
+                Math_ApproachF(&this->fogNear, 0.0f, 1.0f, this->subCamAtVel * 100.0f);
                 Math_ApproachF(&this->subCamAtVel, 2.0f, 1.0f, 0.01f);
             }
 
-            if (this->unk_1D14 == 42) {
-                this->unk_1D78 = 1;
+            if (this->giantsMaskCsTimer == 42) {
+                this->giantsMaskCsFlashState = GIANTS_MASK_CS_FLASH_STATE_STARTED;
             }
 
-            if (this->unk_1D14 > 50) {
-                D_809E0421 = 1;
-                goto block_38;
+            if (this->giantsMaskCsTimer > 50) {
+                sCanSkipMaskOffCs = true;
+                goto done;
             }
             break;
 
-        case 11:
-            if (this->unk_1D14 < 8) {
-                break;
+        case GIANTS_MASK_CS_STATE_MASK_OFF_SKIPPED:
+            if (this->giantsMaskCsTimer >= 8) {
+                goto done;
             }
+            break;
 
-        block_38:
-        case 20:
-            this->unk_1D18 = 0;
+        done:
+        case GIANTS_MASK_CS_STATE_DONE:
```

Comment:
```suggestion
        case GIANTS_MASK_CS_STATE_DONE:
        done:
```
Possibly just a personal preference, but it kind of think the case label should go first (if it matches of course).

Reply:
Doesn't match unfortunately :( What's funny is that the case label isn't even necessary; nothing actually sets `giantsMaskCsState` to `GIANTS_MASK_CS_STATE_DONE`. But it still needs to exist, and the label still needs to be *before* the case label for it to match.

Reply:
![despair](https://github.com/zeldaret/mm/assets/7416381/652d6fdf-78ef-48cf-ba4f-944952431864)